### DISCUSSION TITLE
AXIS P8815-2 3D People Counter - GET & POST for ocupancy

### DISCUSSION THREAD

**Discussion by alexandrududuman** [AxisCommunications/discussions#84] [General] [answered]
Hi guys, 
I'm hoping to find someone who can help me. 
I have a people counter from AXIS, model  AXIS P8815-2 3D.

A)
I found this request for getting ocupancy:
http://MY_HOST/a3dpc/api/occupancy . 

Indeed, I get what I need:
`{"occupancy":0,"serial":"ac:cc:8e:fc:d0:85","timestamp":"2022-10-19T17:26:07+03:00","total_in":54,"total_out":56}`
This one is working fine from browser. 
However, I would need this request to be able to do it from python code and I need some authentication header i think. Do you know what I need to put and where I can find the credentials I need? In what format?

B)
I found several requests (POST) to change the occupancy. However, none of those tested worked on my model.

MYHOST/axis-cgi/a3dpc.cgi , with this body:
```
{
"apiVersion": "0.5",
"context": "95eb9c7a-5b30-4b78-aafd-eeaec4e87886",
"method": "adjustOccupancy",
"params": {
"occupancy": 0
}
}

```
i received:

```
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html>

<head>
<title>401 Unauthorized</title>
</head>

<body>
<h1>Unauthorized</h1>
<p>This server could not verify that you
are authorized to access the document
requested. Either you supplied the wrong
credentials (e.g., bad password), or your
browser doesn't understand how to supply
the credentials required.</p>
</body>

</html>

```
First of all, is the request correct?  Do I only need that part of the authentication correct?

Can anyone help me with some complete examples for the 2 problems? For GET & POST occupancy :)

**Comment by alexandrududuman** [comment, marked as answer]
I looked at some previous threads, I made it, I'm posting here for others who may be in the same situation.

get.py:

```
import requests
import json
from requests.auth import HTTPDigestAuth

cam_user = 'root'
cam_pass = 'pass'
auth=HTTPDigestAuth(cam_user, cam_pass)

url = "http://HOST/a3dpc/api/occupancy"

response = requests.get(url, auth=auth)

print(response.status_code)
print(response.text)
```


post.py:

```
import requests
import json
from requests.auth import HTTPDigestAuth

cam_user = 'root'
cam_pass = 'pass'
auth=HTTPDigestAuth(cam_user, cam_pass)

url = "http://HOST/axis-cgi/a3dpc.cgi"

payload = {
"apiVersion": "0.1",
"context": "95eb9c7a-5b30-4b78-aafd-eeaec4e87886",
"method": "adjustOccupancy",
"params": {
"occupancy": 100
}
}


response = requests.post(url, auth=auth,json=payload)

print(response.status_code)
print(response.text)
```

**Reply by mattias-kindborg-at-work** [reply]
@alexandrududuman, thanks a lot for not only asking the question, but more importantly, also sharing the answer with us in the community. You're a champ!
